container/lxd/container.go

```diff
@@ -66,6 +67,37 @@ func (c *ContainerSpec) ApplyConstraints(serverVersion string, cons constraints.
 		}
 		c.Config["limits.memory"] = fmt.Sprintf(template, *cons.Mem)
 	}
+	if cons.HasArch() {
+		c.Architecture = *cons.Arch
+	}
+	if cons.HasRootDisk() || cons.HasRootDiskSource() {
+		if !cons.HasRootDiskSource() {
+			return errors.New("root disk size constraints require a root disk source")
```

Comment:
@manadart we should change this to not returning an error, instead we should set this to default.